### DISCUSSION TITLE
Should we release 0.72.1?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#74] [Patches]
## Released: refer to #76

Current Release: [0.72.0](https://github.com/facebook/react-native/releases/tag/v0.72.0)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of issues we're aware of

* ~~rn-diff-purge pipeline [is failing](https://github.com/react-native-community/rn-diff-purge/actions/runs/5336160964), no diff on upgrade-helper~~
  * actively getting investigated, tentative fix CLI side available (https://github.com/react-native-community/cli/pull/1980) -> fixed ✅
* https://github.com/facebook/react-native/issues/38012
  * workaround available, waiting on socketrocket to release patched version
* https://github.com/facebook/react-native/issues/38006
* https://github.com/facebook/react-native/issues/37871
* https://github.com/facebook/react-native/issues/37944

#### List of qualified picks

1. ✅ https://github.com/facebook/react-native/commit/377a8b70eee4b1d9f6e641ceda523ebe02c2e92c
2. ✅ https://github.com/facebook/react-native/commit/7a2a3278d08b13dbde7a6e967474c20d6a5c76a5
3. 🦘https://github.com/facebook/react-native/commit/247da6ef7fd0ae335ffc208218532becd596f855 (merge conflict - skipping)
4. ✅ https://github.com/facebook/react-native/commit/a28881a3d79e732670157638aa5207c88c79718c
5. ✅ https://github.com/facebook/react-native/commit/a702d0515f9005714da52cda7f6851e06b4103da
6. ✅ https://github.com/facebook/react-native/commit/ebbd22c9d84800c11715662e2342cd6a7daae848
7. ✅ https://github.com/facebook/react-native/commit/bbcedd385bf7fe374955378d2c2a065318f740cb
8. ✅ https://github.com/facebook/react-native/commit/83d7a48a46c00b99c52a8ac5897c013924e10152
9. plus, local work:
    1. ✅ reimplement https://github.com/facebook/react-native/pull/37869
    2. ✅ bump CLI to 11.3.3
    3. 🟡 bump socketrocket once new release is out: https://github.com/facebookincubator/SocketRocket/commits/main
      * socketRocket 0.7.0 contains some incompatible bumps, so we need the maintainer to release 0.6.1 with just the hotfix that we need.

not merged at the time of release:
* https://github.com/facebook/react-native/pull/35803
* https://github.com/facebook/react-native/pull/38126

#### Local commits to backport to main

1. https://github.com/facebook/react-native/commit/8ce471e2fa802cc50ff2d6ab346627cb5f6d79b4

**Comment by efstathiosntonas** [comment]
Just a heads up, react native upgrade helper does not contain 0.72, only the RC ones

**Reply by leotm** [reply]
same here also checked https://react-native-community.github.io/upgrade-helper/?from=0.71.11&to=0.72.0

**Reply by rnkdsh** [reply]
+1

**Reply by kelset** [reply]
yeah we are aware - we've been trying to investigate it right after 0.72.0 landed and it's a problem with the script that generates the diff ([the pipeline fails](https://github.com/react-native-community/rn-diff-purge/actions/runs/5336160964)). It looks like it has something to do with CLI, a tentative fix is here (https://github.com/react-native-community/cli/pull/1980) and we're trying to dig more.

Because of how our release process works, luckily, there are no changes between last RC and .0, so you can leverage the RC6 diff for upgrading your app while we address this - I've already added a note in the GH release mentioning this.

**Reply by efstathiosntonas** [reply]
@kelset the [CHANGELOG](https://github.com/facebook/react-native/blob/main/CHANGELOG.md) has grown really big thus rendering it unreadable (This blob took too long to generate), do you think you should split it? Had to download the file and enter it on an .md reader to read it, reading the raw requires Superman vision lol

**Reply by kelset** [reply]
oh lol that's a new one 🤣 yeah we can try to chop it down, @lunaleaps was advocating for that too. I'll try to prep something up.

**Reply by kelset** [reply]
PR is up https://github.com/facebook/react-native/pull/38017

**Reply by kelset** [reply]
fixed -> https://react-native-community.github.io/upgrade-helper/?from=0.71.11&to=0.72.0

**Reply by kelset** [reply]
CLI got a release with the fix, we need the bump to CLI 11.3.3 -> https://github.com/react-native-community/cli/commits/11.x

**Comment by johnf** [comment]
I haven't gotten to the bottom of it yet but 
https://github.com/facebook/react-native/pull/37918/commits/a2b31dd1f8194c151e283fdbf5481eae61413cb1 might be a good candidate.

I've upgraded to 0.72.0 today. I hit an issue where`performance.now` was not defined. I pulled in the changes in the above and associated commits which solved that problem.

But now 'setTimeout' is not defined. It works if I create a new project so one of my deps must be causing issues. I'll update as I discover more.

**Reply by johnf** [reply]
Ignore this. I had `react-native-codegen: ^0.71.0` in package.json. Not sure why I had it there at all as it's not needed

**Comment by efstathiosntonas** [comment]
Can you please merge this [PR](https://github.com/facebookincubator/SocketRocket/pull/617/files) for `SocketRocket`?  

It solves this issue https://github.com/facebook/react-native/issues/38012

**Reply by cipolleschi** [reply]
working on it. The PR has been merged, but the maintainer would like to do some other work and will try to release a new version of SoketRocket early next week.

**Reply by efstathiosntonas** [reply]
@cipolleschi hi, @nlutsenko has released [0.7.0](https://github.com/facebookincubator/SocketRocket/releases/tag/0.7.0), just a heads up. Thanks @nlutsenko

**Reply by kelset** [reply]
@efstathiosntonas sadly that release also contains some incompatible bumps, so we need the maintainer to release 0.6.1 with just the hotfix that we need. It won't make it in time for 0.72.1 but we'll handle it for 0.72.2

**Reply by kelset** [reply]
plottwist: 0.6.1 has already been released so we should be able to include it

**Comment by efstathiosntonas** [comment]
Love the silence after the 1000 commits storm 😂

**Comment by Bardiamist** [comment]
Updated to 0.72.0 and have this problem: https://github.com/reactwg/react-native-releases/discussions/54#discussioncomment-5984629

Looks this not helped for build from Android Studio: https://github.com/facebook/react-native/pull/37869/files
Maybe need to update `react-native-gradle-plugin`

Helped only add `kotlin.jvm.target.validation.mode=warning` into `gradle.properties`

**Reply by efstathiosntonas** [reply]
I've faced that too, on `android/build.gradle` add this at the very bottom: 

```gradle
// Workaround: Force all android libraries to use java 11.
// This is to fix the build errors from some third party libraries still using java 8,
// which cause gradle 8 errors:
// 'compileDebugJavaWithJavac' task (current target is 1.8) and 'compileDebugKotlin' task (current target is 11) jvm target compatibility should be set to the same Java version.
allprojects { subproject ->
    subproject.pluginManager.withPlugin("com.android.library") {
        subproject.extensions.findByType(com.android.build.api.variant.AndroidComponentsExtension)?.finalizeDsl { ext ->
            ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_11
            ext.compileOptions.targetCompatibility = JavaVersion.VERSION_11
        }
    }
}
```

and on `android/app/build.gradle` add this inside the `android {`:

```gradle
android {
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
    }

    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_11.majorVersion
    }
...
```

and on the same file add this on top right under `apply plugin: "com.facebook.react"`
`apply plugin: "kotlin-android"`

**Reply by Bardiamist** [reply]
React native 0.72 lands with @react-native/gradle-plugin@0.72.10 https://github.com/facebook/react-native/blob/0.72-stable/packages/react-native/package.json#L87

When this change https://github.com/facebook/react-native/pull/37869/files in 0.73.0

**Reply by cortinico** [reply]
Hey all,
This is a known issue.
The correct fix is:
- https://github.com/facebook/react-native/pull/37869
This will be included in 0.72.1

As a workaround for everyone facing this same issue right now, you can add:
```
kotlin.jvm.target.validation.mode=warning
```

to your `android/gradle.properties` file.

**Reply by kelset** [reply]
yes, that change will make it to 0.72.1 - it landed after RC6 so we had to revert the local commit, we'll add it back in 0.72.1 - which will likely be out this week. Depends on status of other open issues

**Comment by cipolleschi** [comment]
Potentially, for the interop layer:
- https://github.com/facebook/react-native/pull/38089
- https://github.com/facebook/react-native/pull/38093

**Comment by huntie** [comment]
Please pick:
- https://github.com/facebook/react-native/pull/38092

**Comment by bernhardoj** [comment]
Hi, can we CP this commit, please:

- https://github.com/facebook/react-native/commit/83d7a48a46c00b99c52a8ac5897c013924e10152

**Comment by huntie** [comment]
I'm landing this change (to pick, please 🙏🏻) and hopefully we also get a CLI bump in today. **Non-blocking for 0.72.1** (is an enhancement).

- https://github.com/facebook/react-native/pull/38126

**Reply by kelset** [reply]
added to the list for 0.72.2 👍
